src/axom/core/ArrayBase.hpp

```diff
@@ -552,6 +648,13 @@ class ArrayBase<T, 1, ArrayType>
     return idx >= 0 && idx < asDerived().size();
   }
   /// @}
+protected:
+  /// \brief The sizes (extents?) in each dimension
+  StackArray<IndexType, 1> m_dims;
+  /// \brief Spacing between consecutive elements
+  IndexType m_spacing;
+  /// \brief The strides in each dimension
+  StackArray<IndexType, 1> m_strides;
```

Comment:
Is there a reason that two of these are `StackArray`s and the other is a scalar? E.g. a host-device/memory thing? Or is it purely to mirror the multidimensional case?

Reply:
Purely to mirror the multidimensional case.  I wasn't considering the memory overhead and didn't fully appreciate the performance implications.